### ISSUE TITLE
Run fails to complete after successful run with test data

### ISSUE THREAD

**Issue by alexjfortuna** [phac-nml/snvphyl-galaxy-cli#2]
Hi, my run is failing to complete when attempting to upload files to galaxy. I have tried both with --fastq-files-as-links and without. Also --with-docker-sudo and without. These are fastq files I have used for other analysis, same as the reference. I am a bit stuck for troubleshooting, any comments appreciated. Thank you. 

Log:

python bin/snvphyl.py --with-docker-sudo --deploy-docker --fastq-dir example-data/Listeria-WGS-test --reference-file example-data/GCF_EGDe.fasta --min-coverage 5 --output-dir output3

Deploying Docker Container
==========================
Running 'sudo docker run --detach --publish 48888:80 --volume /home/afortuna/Applications_directory/snvphyl-galaxy-cli/example-data/Listeria-WGS-test:/snvphyl-data/fastq phacnml/snvphyl-galaxy-1.0.1:1.0.1b'
Docker id c0d07c6c9bf7f81c9e9c8fc09cecfb14a7a8029240613a0cc6aa403eaf384c62
Waiting for docker to complete launching Galaxy on port 48888 ......finished.  Galaxy in Docker has (hopefully) started successfully.
Took 0.61 minutes to deploy docker

Examining input fastq files
===========================
Structuring data in directory 'example-data/Listeria-WGS-test' like:
00-0344: paired {forward: example-data/Listeria-WGS-test/00-0344_R1.fastq, reverse: example-data/Listeria-WGS-test/00-0344_R2.fastq}
00-0345: paired {forward: example-data/Listeria-WGS-test/00-0345_R1.fastq, reverse: example-data/Listeria-WGS-test/00-0345_R2.fastq}

Set up workflow input
=====================
setting parameter {irida.corefacility.ca/galaxy-shed/repos/nml/consolidate_vcfs/consolidate_vcfs/1.8.0, snv_abundance_ratio, 0.75}
setting parameter {irida.corefacility.ca/galaxy-shed/repos/nml/consolidate_vcfs/consolidate_vcfs/1.8.0, coverage, 5}
setting parameter {irida.corefacility.ca/galaxy-shed/repos/nml/verify_map/verify_map/1.8.0, mindepth, 5}
setting parameter {irida.corefacility.ca/galaxy-shed/repos/nml/consolidate_vcfs/consolidate_vcfs/1.8.0, mean_mapping, 30}
setting parameter {irida.corefacility.ca/galaxy-shed/repos/nml/find_repeats/findrepeat/1.8.0, length, 150}
setting parameter {irida.corefacility.ca/galaxy-shed/repos/nml/find_repeats/findrepeat/1.8.0, pid, 90}
setting parameter {irida.corefacility.ca/galaxy-shed/repos/nml/consolidate_vcfs/consolidate_vcfs/1.8.0, use_density_filter.threshold, 2}
setting parameter {irida.corefacility.ca/galaxy-shed/repos/nml/consolidate_vcfs/consolidate_vcfs/1.8.0, use_density_filter.window_size, 500}

Upload files to Galaxy
======================
Creating history in Galaxy name 'snvphyl-GCF_EGDe-2017-12-19-run'
Uploading reference file example-data/GCF_EGDe.fasta
Uploading fastq files in history 'snvphyl-GCF_EGDe-2017-12-19-run'
Uploading as link example-data/Listeria-WGS-test/00-0344_R1.fastq
Uploading as link example-data/Listeria-WGS-test/00-0344_R2.fastq
Uploading as link example-data/Listeria-WGS-test/00-0345_R1.fastq
Uploading as link example-data/Listeria-WGS-test/00-0345_R2.fastq
Waiting for library datasets to finish processing...4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.4.
Undeploying and cleaning up Docker Container
=============================================
Running 'sudo docker rm -f -v c0d07c6c9bf7f81c9e9c8fc09cecfb14a7a8029240613a0cc6aa403eaf384c62'
c0d07c6c9bf7f81c9e9c8fc09cecfb14a7a8029240613a0cc6aa403eaf384c62
Traceback (most recent call last):
  File "bin/snvphyl.py", line 1286, in <module>
    main(snvphyl_version_settings, **dic)
  File "bin/snvphyl.py", line 961, in main
    repeat_minimum_length, repeat_minimum_pid, filter_density_window, filter_density_threshold, invalid_positions_file, output_dir)
  File "bin/snvphyl.py", line 1071, in main_galaxy
    fastq_id=upload_fastq_collection_paired(gi,history_id,fastq_paired)
  File "bin/snvphyl.py", line 568, in upload_fastq_collection_paired
    paired_elements=upload_fastqs_library_paired(gi,history_id,created_library['id'],fastq_paired)
  File "bin/snvphyl.py", line 477, in upload_fastqs_library_paired
    fastq_history_ids=upload_fastqs_to_history_via_library(gi,history_id,library_id,fastqs_to_upload)
  File "bin/snvphyl.py", line 430, in upload_fastqs_to_history_via_library
    raise Exception("Error uploading fastq file ("+fastq_library_ids[dataset]+", "+dataset+") to Galaxy Library "+library_id)
Exception: Error uploading fastq file (00-0344/forward, f2db41e1fa331b3e) to Galaxy Library f2db41e1fa331b3e

**Comment by apetkau** [comment]
Could you let me know the permissions for the files in `example-data/Listeria-WGS-test/`?  That is running:

```bash
ls -l example-data/Listeria-WGS-test/
```

If only your user has read permission on the files (`-rw-------`) you may need to add read permission for everyone `chmod +r example-data/Listeria-WGS-test/`.  Since this is running through Docker, the user running Docker is different from your regular user, and so if files are uploaded as links to the originals inside of Docker then Galaxy (in Docker) will not have read permissions on the files.

Alternativly, you can use `--copy-fastq-files-to-docker` to make copies of the files into Docker first (avoiding the permissions issues).  I think this is what you were trying to do when running with and without --fastq-files-as-links, but it defaults to making fastq files as links when running via Docker (sorry for the confusingly named parameters).

If this doesn't work, then I would ask if you could try out running without 00-0344 (that is, does it only occur with 00-0344 or does it occur with every fastq file).

I hope this helps you,

Aaron

**Comment by alexjfortuna** [comment]
Hi Aaron,

Thank you for getting back to me. The permissions for the files are
-rwxrwx--- I added read permissions to all and it worked! 

However;

snvphyl.py: error: unrecognized arguments: --copy-fastq-files-to-docker, I was able to execute with --fastq-files-as-links  in an attempt to cut down on processing time

Thank you,

Alex

**Comment by apetkau** [comment]
Awesome :smile: 

For `--copy-fastq-files-to-docker` you likely have to checkout the `development` branch.  That is:

```
cd snvphyl-galaxy-cli
git checkout development
```

I haven't made a release with those parameters yet (sorry about that).

**Comment by apetkau** [comment]
This code was finally released in version `1.4.0` - https://github.com/phac-nml/snvphyl-galaxy-cli/releases/tag/1.4.0
